src/commands/dataconnect-execute.ts

```diff
@@ -0,0 +1,306 @@
+        const url = new URL("http://placeholder");
+        url.host = emulatorHost;
+        apiClient = new Client({
+          urlPrefix: url.toString(),
+          apiVersion: DATACONNECT_API_VERSION,
+        });
+      } else {
+        await requireAuth(options);
+        apiClient = dataconnectDataplaneClient();
+      }
+
+      if (!serviceName) {
+        if (serviceId && (locationId || emulatorHost)) {
+          serviceName = `projects/${projectId}/locations/${locationId || "unused"}/services/${serviceId}`;
+        } else {
+          serviceName = (await getServiceInfo()).serviceName;
+        }
+      }
+      if (!options.vars && !process.stdin.isTTY && !stdinUsedFor) {
+        options.vars = "@-";
+      }
+      const unparsedVars = await literalOrFile(options.vars, "--vars");
+      const unparsedClaims = await literalOrFile(options.impersonate, "--impersonate");
+      let impersonate: Impersonation | undefined = undefined;
+      if (unparsedClaims === "unauthenticated") {
+        impersonate = { unauthenticated: true, includeDebugDetails: !options.noDebugDetails };
+      } else if (unparsedClaims) {
+        impersonate = {
+          authClaims: parseJsonObject(unparsedClaims),
+          includeDebugDetails: !options.noDebugDetails,
+        };
+      }
+
+      const response = await executeGraphQL(apiClient, serviceName, {
+        query,
+        operationName,
+        variables: parseJsonObject(unparsedVars),
+        extensions: { impersonate },
+      });
+
+      // If the status code isn't OK or the top-level `error` field is set, this
+      // is an HTTP / gRPC error, not a GQL-compatible error response.
+      let err = responseToError(response, response.body);
+      if (isGraphQLResponseError(response.body)) {
+        const { status, message } = response.body.error;
+        if (!err) {
+          err = new FirebaseError(message, {
+            context: {
+              body: response.body,
+              response: response,
+            },
+            status: response.status,
+          });
+        }
+        if (status === "INVALID_ARGUMENT" && message.includes("operationName is required")) {
+          throw new FirebaseError(
+            err.message + `\nHint: Append <operationName> as an argument to disambiguate.`,
+            { ...err, original: err },
+          );
+        }
+      }
+      if (err) {
+        throw err;
+      }
+
+      // If we reach here, we should have a GraphQL response with `data` and/or
+      // `errors` (note the plural). First let's double check that's the case.
+      if (!isGraphQLResponse(response.body)) {
+        throw new FirebaseError("Got invalid response body with neither .data or .errors", {
+          context: {
+            body: response.body,
+            response: response,
+          },
+          status: response.status,
+        });
+      }
+
+      // Log the body to stdout to allow pipe processing (even with .errors).
```

Comment:
This doesn't work quite nicely since errors thrown (below, if any) are also logged to stdout (globally handled). Piping to `jq` for example can choke on the extra output